### ISSUE TITLE
Missing .el in build (embark-org.el missing)

### ISSUE THREAD

**Issue by ahmed-shariff** [radian-software/straight.el#1046]
After doing a pull-all, for some reason the build is excluding the `embark-org.el` file (and it's corresponding elc file).

Surprisingly, with the `straigh-report-bug` macro, the resulting build has the `embark-org.el` file:

<details open><summary>Test Case</summary>

```emacs-lisp
(straight-bug-report
  :pre-bootstrap 
  (message "before bootstrap")
  :post-bootstrap 
  (setq straight-use-package-by-default t package-enable-at-startup nil straight-recipes-gnu-elpa-use-mirror t straight-vc-git-default-protocol 'https straight-host-usernames
        '((github . "ahmed-shariff")))
  (message "after bootstrap")
  (straight-use-package
   '(use-package))

  ;; The full emabrk config I use, for good measure
  (use-package embark :bind
    (("C-." . embark-act)
     ("C-;" . embark-dwim)
     ("C-h B" . embark-bindings))
    :init
    (setq prefix-help-command #'embark-prefix-help-command)
    :config
    (add-to-list 'display-buffer-alist
                 '("\\`\\*Embark Collect \\(Live\\|Completions\\)\\*" nil
                   (window-parameters
                    (mode-line-format . none))))
    (define-key embark-file-map "o" nil)
    (define-key embark-file-map "ocn" #'copy-buffer-file-name)
    (define-key embark-file-map "ocf" #'copy-file-full-path)
    (defun embark-which-key-indicator nil "An embark indicator that displays keymaps using which-key.\nThe which-key help message will show the type and value of the\ncurrent target followed by an ellipsis if there are further\ntargets."
           (lambda
             (&optional keymap targets prefix)
             (if
                 (null keymap)
                 (which-key--hide-popup-ignore-command)
               (which-key--show-keymap
                (if
                    (eq
                     (plist-get
                      (car targets)
                      :type)
                     'embark-become)
                    "Become"
                  (format "Act on %s '%s'%s"
                          (plist-get
                           (car targets)
                           :type)
                          (embark--truncate-target
                           (plist-get
                            (car targets)
                            :target))
                          (if
                              (cdr targets)
                              "…" "")))
                (if prefix
                    (pcase
                        (lookup-key keymap prefix 'accept-default)
                      ((and
                        (pred keymapp)
                        km)
                       km)
                      (_
                       (key-binding prefix 'accept-default)))
                  keymap)
                nil nil t
                (lambda
                  (binding)
                  (not
                   (string-suffix-p "-argument"
                                    (cdr binding))))))))
    (setq embark-indicators
          '(embark-which-key-indicator embark-highlight-indicator embark-isearch-highlight-indicator))
    (defun embark-hide-which-key-indicator
        (fn &rest args)
      "Hide the which-key indicator immediately when using the completing-read prompter."
      (which-key--hide-popup-ignore-command)
      (let
          ((embark-indicators
            (remq #'embark-which-key-indicator embark-indicators)))
        (apply fn args)))
    (advice-add #'embark-completing-read-prompter :around #'embark-hide-which-key-indicator))
  (message "%s"
           (directory-files
            (format "%s/straight/repos/embark" straight-base-dir)))
  (message "%s"
           (directory-files
            (format "%s/straight/build/embark" straight-base-dir)))
  :preserve t)
```
</details>

- Test run at: `2023-01-31 20:13:01`
- system-type: `windows-nt`
- straight-version: `prerelease
 (HEAD -> master, origin/master) 3eca39d 2022-11-25`
- emacs-version: `GNU Emacs 28.1 (build 2, x86_64-w64-mingw32)
 of 2022-04-21`

<details><summary>Output</summary>

```emacs-lisp
before bootstrap
Bootstrapping straight.el...
Bootstrapping straight.el...done

Test run with version: prerelease
 (HEAD -> develop, origin/develop) 3fa415c 2023-01-20
after bootstrap

(. .. .dir-locals.el .elpaignore .git .gitignore LICENSE README.org avy-embark-collect.el embark-consult.el embark-org.el embark.el embark.texi)
(. .. embark-autoloads.el embark-org.el embark-org.elc embark.el embark.elc embark.texi)
Packages:
"straight"                n/a                  develop 3fa415c 2023-01-20
"org-elpa"                n/a                  n/a
"melpa"                   n/a                  master cd55f4fb 2023-01-30
"gnu-elpa-mirror"         n/a                  master eb11995 2022-12-15
"nongnu-elpa"             n/a                  main 85c7a060c6 2023-01-26
"el-get"                  melpa                master 17db7640 2023-01-28
"emacsmirror-mirror"      n/a                  master a2d65d0 2023-01-11
"use-package"             melpa                master bcf0984 2022-12-09
"bind-key"                melpa                master bcf0984 2022-12-09
"embark"                  melpa                master 127492f 2023-01-31
"compat"                  gnu-elpa-mirror      master 8885b1c 2023-01-13

Looking for gnu-elpa-mirror recipe -> Cloning melpa...
Looking for gnu-elpa-mirror recipe -> Cloning melpa...done
Looking for nongnu-elpa recipe -> Cloning gnu-elpa-mirror...
Looking for nongnu-elpa recipe -> Cloning gnu-elpa-mirror...done
Looking for emacsmirror-mirror recipe -> Cloning nongnu-elpa...
Looking for emacsmirror-mirror recipe -> Cloning nongnu-elpa...done
Looking for emacsmirror-mirror recipe -> Cloning el-get...
Looking for emacsmirror-mirror recipe -> Cloning el-get...done
Looking for straight recipe -> Cloning emacsmirror-mirror...
Looking for straight recipe -> Cloning emacsmirror-mirror...done
Building straight...
Building straight...done
Cloning use-package...
Cloning use-package...done
Building use-package...
Building use-package -> Building bind-key...
Building use-package -> Building bind-key...done
Building use-package...
Building use-package...done
Cloning embark...
Cloning embark...done
Building embark...
Building embark -> Cloning compat...
Building embark -> Cloning compat...done
Building embark -> Building compat...
Building embark -> Building compat...done
Building embark...
Building embark...done

```
</details>

Where as the build directory on system has only the following:
<img width="126" alt="image" src="https://user-images.githubusercontent.com/15046660/215951872-4c8df7df-51c7-4386-a159-4c987cc10678.png">

I am stumped on how to debug this. Would appreciate any help.

**Comment by ahmed-shariff** [comment]
Not sure if this is some bizzare edge case. I deleted the `build` dir and restarted emacs. That seems to have fixed it? Closing for now.

**Comment by WorldsEndless** [comment]
I just did some upgrades (including embark) and have the same error. I notice that the build does not include the embark-org.el but the repo definitely does.  As a work around I have been navigating to the repo directory and evaluating the embark-org file and the errors get out of my way until the next restart. I'll try deleting the build dir for embark and see if that does the trick.

**Comment by WorldsEndless** [comment]
I have tried removing the build directory and when it rerceates the directory the problem is still there: embark-org is missing. I have tried symlinking embark-org manually into the straight build directory, but  the symlink gets wiped out upon re-build of straight. I tried setting the chattr to make the file undeletable, but that can't be done on links. In anycase, the correct fix would just be to figure out how to get Straight to include embark-org.el in its build.
